src/chat/zulip.md

```diff
@@ -0,0 +1,84 @@
+## Zulip
+
+[Rust's Zulip](rust-lang.zulipchat.com) is used by a number of teams, notably the compiler and language
+teams, along with their working groups.
+
+Zulip can be an unintuitive platform to get started with. The UI is divided into
+two areas: a list of streams on the left, and the main chat in the middle.
+
+## Streams
+
+These are similar to "channels" on other platforms (i.e., there should not be
+too many). On the other hand, you can choose which streams you subscribe to, so
+there can be more than channels on other platforms.
+
+Streams are appropriate for any Rust official group. For example, working
+groups, project groups, teams are all examples of official groups. These should
+ideally also be represented in the [team repository](../infra/team-maintenance.md).
+
+A stream should be named such as `#t-{team}/{group name}`. For example,
+`#t-compiler/wg-parallel-rustc`.
+
+If no top-level team exists, or the group spans multiple teams (e.g.,
+project-ffi-unwind), then the top level team should be omitted.
+
+It is not generally appropriate to have groups that do not correspond to a set
+of people. For example, `#t-compiler/help` would not be appropriate, as such
+discussion should go into `#t-compiler` (the catch-all compiler stream).
+
+When a new stream is started, you should announce it in `#announce`. This is
+generally done automatically by Zulip.
+
+## Topics
+
+One of these is attached to every message within a given stream (these are the
+subdivisions within streams). Topics are generally transient, and live for as
+long as there is active discussion on a topic.
+
+New conversation in a given stream should almost always start in a new topic,
+not a preexisting one. Unlike (for example) GitHub issues, you should not
+attempt to search for a past topic on the same subject. Do not spend too long on
+the name of the topic, either, beyond trying to make it short. It is similar in
+nature to the subject line of an email, but should generally be no longer than
+20 characters (loosely two to three words), to make sure it is visible to users.
+
+You should eagerly fork new discussion topics into fresh topics. Note that this
+can be done with the tail of another topic (if accidentally you diverge into
+another area of discussion).
+
+To fork from an existing topic, find the first message that should go into the
+new topic. All subsequent messages will be sent to the new topic.
+
+Choose the down arrow in the upper right of the message box, and click
+"View source / Edit topic".
+
+![Screenshot highlighting the "View Source / Edit topic" option from the down
+arrow on Zulip.](./images/zulip-view-source.png)
+
+Then, edit the topic message inline and choose "Change later messages to
+this topic".
+
+![Screenshot highlighting the "Change later messages to this topic" option from
+the topic change action menu.](./images/zulip-change-later-messages.png)
```

Comment:
Maybe worth commenting that if a topic is related to a Github issue or PR it's nice to include #issue_number so you can click it from the topic and jump to the corresponding issue.

Reply:
Oh, yes, we could document the "linkifiers" we support (and mention that we can add new ones). e.g., `#123`, `rust-lang/foo#456`, etc. I'm actually not sure what the full set of linkifiers are.

Reply:
You can find the full set by going to "Organisation settings" -> "Linkifiers" (should probably also mention that). This is the current set we have.
<img width="749" alt="Screenshot 2019-12-30 at 14 33 08" src="https://user-images.githubusercontent.com/4464295/71586232-6492c480-2b11-11ea-84ab-989c9909cfa9.png">